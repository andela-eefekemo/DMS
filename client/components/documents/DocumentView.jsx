import React from 'react';
import { Modal } from 'react-materialize';
import ReactHtmlParser from 'react-html-parser';
import InputField from '../common/InputField';
import Dropdown from '../common/Dropdown';

const DocumentView = (props) => {
  const {
    id,
    title,
    content,
    access, onSubmit, onChange, deleteDocument, userId, authorId } = props;
  return (
    <div className="card blue-grey darken-1">
      <div className="card-content white-text">
        <span className="card-title">{title}</span>
        <div>{ReactHtmlParser(content)}</div>
        <p>{access}</p>
      </div>
      <div className="card-action">
        <div className="row">
          <div className="col l6 m6 s12">
            {userId === authorId && <button
              className="waves-effect btn button-design"
              onClick={deleteDocument}
              value={id}
            >
              Delete
      </button>
            }
          </div>
          <div className="col l6 m6 s12">
            {userId === authorId && <Modal
              trigger={
                <a
                  className="waves-effect btn button-design"
                  data-target="passwordModal">
                  Update
          </a>
              }
            >
              <form onSubmit={onSubmit} >
                <h5>Create Document</h5>
                <InputField
                  name="title"
                  value={title}
                  placeholder="Document Title"
                  className="validate form-design"
                  type="text" onChange={onChange} />
                <InputField
                  name="content"
                  value={content}
                  placeholder="Document Content"
                  className="validate form-design"
                  type="text" onChange={onChange} />
                <select
                  name="access"
                  className="browser-default input-field select" onChange={onChange}>
                  <Dropdown value="" text="Select access type" />
                  <Dropdown value="public" text="Public" />
                  <Dropdown value="private" text="Private" />
                  <Dropdown value="role" text="Role" />
                </select>
                <div className="input-field center">
                  <button className="waves-effect btn button-design" type="submit">
                    Update
            </button>
                </div>

              </form>
            </Modal>}
          </div>
        </div>
      </div>
    </div>

  );
};

export default DocumentView;
