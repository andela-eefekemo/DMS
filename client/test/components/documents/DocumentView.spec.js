/* global expect jest test */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DocumentView from '../../../components/documents/DocumentView';

jest.dontMock('../../../components/documents/DocumentView');

describe('DocumentView Component', () => {
  test('should match the document view snapshot', () => {
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const deleteDocument = jest.fn();
    const component = shallow(
      <DocumentView
        title={'Book'}
        id={3}
        content={'Awesome'}
        access={'public'}
        onChange={onChange}
        onSubmit={onSubmit}
        deleteDocument={deleteDocument}
        userId={3}
        authorId={3} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
