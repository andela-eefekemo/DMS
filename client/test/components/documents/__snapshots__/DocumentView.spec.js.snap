// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentView Component should match the document view snapshot 1`] = `
<div
  className="card blue-grey darken-1"
>
  <div
    className="card-content white-text"
  >
    <span
      className="card-title"
    >
      Book
    </span>
    <div>
      Awesome
    </div>
    <p>
      public
    </p>
  </div>
  <div
    className="card-action"
  >
    <div
      className="row"
    >
      <div
        className="col l6 m6 s12"
      >
        <Modal
          actions={
            <div>
              <button
                className="btn btn-flat modal-action modal-close"
                id="delete"
                name={3}
                onClick={[Function]}
              >
                Delete
              </button>
              <button
                className="btn btn-flat modal-action modal-close left"
              >
                Cancel
              </button>
            </div>
          }
          bottomSheet={false}
          fixedFooter={false}
          modalOptions={Object {}}
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
              id="delete-document"
            >
              Delete
            </a>
          }
        >
          <div>
            <h5
              className="center"
            >
              Are you sure you want to delete the user
            </h5>
          </div>
        </Modal>
      </div>
      <div
        className="col l6 m6 s12"
      >
        <Modal
          actions={
            Array [
              <Button
                flat={true}
                modal="close"
                node="button"
                waves="light"
            >
                Close
            </Button>,
            ]
          }
          bottomSheet={false}
          fixedFooter={false}
          modalOptions={Object {}}
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
              id="update-document"
            >
              Update
            </a>
          }
        >
          <div>
            <h5>
              Create Document
            </h5>
            <InputField
              className="validate form-design"
              name="title"
              onChange={[Function]}
              placeholder="Document Title"
              type="text"
              value="Book"
            />
            <InputField
              className="validate form-design"
              name="content"
              onChange={[Function]}
              placeholder="Document Content"
              type="text"
              value={
                Array [
                  "Awesome",
                ]
              }
            />
            <select
              className="browser-default input-field select"
              name="access"
              onChange={[Function]}
            >
              <Dropdown
                text="Select access type"
                value=""
              />
              <Dropdown
                text="Public"
                value="public"
              />
              <Dropdown
                text="Private"
                value="private"
              />
              <Dropdown
                text="Role"
                value="role"
              />
            </select>
            <div
              className="input-field center"
            >
              <button
                className="modal-action modal-close btn button-design"
                id="save-update"
                onClick={[Function]}
                type="submit"
              >
                Update
              </button>
            </div>
          </div>
        </Modal>
      </div>
    </div>
  </div>
</div>
`;
